### ISSUE TITLE
scalar.js not found asp net core

### ISSUE THREAD

**Issue by rzurlo** [scalar/scalar#3810]
### What happens?

Scalar returns 404 "not found" error if the API is located in a virtual application in IIS and not on the main site. Version 1.2.31. Does anyone have any suggestions on how to fix this? Thanks

### What did you expect to happen?

The same bevahior like api is installed on main site

**Comment by hanspagel** [comment]
Thanks for the report! We’ll look into this.

**Comment by xC0dex** [comment]
Hey @rzurlo,
could you please provide a minimal reproducible project that demonstrates the issue? This would help us a lot. Thanks!

Does everything work as expected when you set the `CdnUrl` to point to a CDN? 

```c#
app.MapScalarApiReference(options =>
{
    options.WithCdnUrl("https://cdn.jsdelivr.net/npm/@scalar/api-reference");
});
```

**Comment by rzurlo** [comment]
Hey @xC0dex,
 with the CDN i have the same issue.
I created an example project in Visual Studio 2022 with .NET 8 API and then published it under a virtual directory local IIS 10 of windows 11, and I get the same exact error. The error is systematic.
Here the sample project : https://github.com/rzurlo/LabScalar

Thanks

**Comment by TobiasMesquitaOMG** [comment]
try that:
```cs
// in my opinion, u would store/retrieve the virtual path from the config (.env/appsettings.json)
var virtualPath = getVirtualPathSomehow(); 
app.MapScalarApiReference(options => 
{
    options.OpenApiRoutePattern = string.IsNullOrWhiteSpace(settings.Api.VirtualPath)
        ? "/openapi/{documentName}.{json|yaml}"
        : $"/{virtualPath}/openapi/{{documentName}}.{{json|yaml}}";
    options.CdnUrl = string.IsNullOrWhiteSpace(settings.Api.VirtualPath)
        ? "/scalar/scalar.js"
        : $"/{virtualPath}/scalar/scalar.js";
    options.EndpointPathPrefix = "/scalar/{documentName}";
});
```

**Comment by TobiasMesquitaOMG** [comment]
try that:

```cs
// in my opinion, u would store/retrieve the virtual path from/to the config (.env/appsettings.json)
var virtualPath = getVirtualPathSomehow(); 
app.MapScalarApiReference(options => 
{
    options.OpenApiRoutePattern = string.IsNullOrWhiteSpace(settings.Api.VirtualPath)
        ? "/openapi/{documentName}.{json|yaml}"
        : $"/{virtualPath}/openapi/{{documentName}}.{{json|yaml}}";
    options.CdnUrl = string.IsNullOrWhiteSpace(settings.Api.VirtualPath)
        ? "/scalar/scalar.js"
        : $"/{virtualPath}/scalar/scalar.js";
    options.EndpointPathPrefix = "/scalar/{documentName}";
});
```

**Comment by TobyMosque** [comment]
try that:
```cs
// in my opinion, u would store/retrieve the virtual path from/to the config (.env/appsettings.json)
var virtualPath = getVirtualPathSomehow(); 
app.MapScalarApiReference(options => 
{
    options.OpenApiRoutePattern = string.IsNullOrWhiteSpace(settings.Api.VirtualPath)
        ? "/openapi/{documentName}.{json|yaml}"
        : $"/{virtualPath}/openapi/{{documentName}}.{{json|yaml}}";
    options.CdnUrl = string.IsNullOrWhiteSpace(settings.Api.VirtualPath)
        ? "/scalar/scalar.js"
        : $"/{virtualPath}/scalar/scalar.js";
    options.EndpointPathPrefix = "/scalar/{documentName}";
});
```

**Comment by rzurlo** [comment]
@TobyMosque  with this  settings it doesn't even work in the mode that previously worked, which is the API published directly as a site.

**Comment by TobyMosque** [comment]
// in my opinion, u would store/retrieve the virtual path from/to the config (.env/appsettings.json)
@rzurlo as soon i cleared my cache, my code stoped to work, so i ended with this one:

```cs
var virtualPath = getVirtualPathSomehow(); 

var docUrl = "api-doc";
var cdnUrl = $"/{docUrl}/scalar.js";
var openApiPattern = "/openapi/{documentName}.{json|yaml}";

var fileProvider = new EmbeddedFileProvider(typeof(ScalarEndpointRouteBuilderExtensions).Assembly, "ScalarStaticAssets");
app.MapGet(cdnUrl, () =>
{
    var resourceFile = fileProvider.GetFileInfo("scalar.js");
    return Results.Stream(resourceFile.CreateReadStream(), "text/javascript", lastModified: resourceFile.LastModified);
});

var res = app.MapScalarApiReference(options =>
{
    options.OpenApiRoutePattern = string.IsNullOrWhiteSpace(settings.Api.VirtualPath)
        ? openApiPattern
        : $"/{virtualPath}{openApiPattern}";
    options.CdnUrl = string.IsNullOrWhiteSpace(settings.Api.VirtualPath)
        ? cdnUrl
        : $"/{virtualPath}{cdnUrl}";
    options.EndpointPathPrefix = $"/{docUrl}/{{documentName}}";
});
```

i got the insight to the code above from here:
https://github.com/scalar/scalar/blob/be5022521e839e61bbcea9b3c125b17064c894a7/packages/scalar.aspnetcore/src/Scalar.AspNetCore/Extensions/ScalarEndpointRouteBuilderExtensions.cs#L58

**Comment by TobyMosque** [comment]
```cs
var virtualPath = "/virtual-path";
var docUrl = "api-doc";
var cdnUrl = $"{docUrl}/scalar.js";
var openApiPattern = "/openapi/{documentName}.{json|yaml}";

app.UseSwagger(opts =>
{
    opts.RouteTemplate = openApiPattern;
});

var fileProvider = new EmbeddedFileProvider(typeof(ScalarEndpointRouteBuilderExtensions).Assembly, "ScalarStaticAssets");
app.MapGet(cdnUrl, () =>
{
    var resourceFile = fileProvider.GetFileInfo("scalar.js");
    return Results.Stream(resourceFile.CreateReadStream(), "text/javascript", lastModified: resourceFile.LastModified);
});

app.MapScalarApiReference(options =>
{
    options.OpenApiRoutePattern = $"{virtualPath}{openApiPattern}";
    options.CdnUrl = $"{virtualPath}/{cdnUrl}";
    options.EndpointPathPrefix = $"/{docUrl}/{{documentName}}";
});
```

I only needed to manually set the virtual path in the `OpenApiRoutePattern` and `CdnUrl`.

At least to me, that is confunsing as hell, a `VirtualPath` or `PublicPath` property would be welcome...
```cs
// that is just a proporsal, don't try...
app.MapScalarApiReference(options =>
{
    options.PublicPath = "/virtual-path";
});
```

**Comment by rzurlo** [comment]
@TobyMosque  This code works for me as well, thank you very much. In my opinion, this is still a workaround
and the issue should be addressed at the core level by the base library, or as you suggest, make a property public where I can set the application's virtual directory path. Very often, APIs are installed as a virtual application and not directly on a site. 
Thanks again.

Raffaele

**Comment by xC0dex** [comment]
Thanks to Docker and Caddy, I was able to set up a similar environment without needing to install IIS.

I made a modification to the library, so the script is now included like this:
```html
<script src="scalar.js"></script>
```
instead of:
```html
<script src="/scalar/scalar.js"></script>
```
This change removes the need for an absolute path `/scalar/scalar.js` and instead uses a relative path `scalar.js`. This adjustment allows the script to load correctly regardless of the app base path, making it more flexible in different hosting environments.

The only additional change needed to get the UI fully functional is to modify the `OpenApiRoutePattern` property as follows:

````c#
const string subPath = "/api";
app.MapScalarApiReference(options =>
{
    options.OpenApiRoutePattern = $"{subPath}{options.OpenApiRoutePattern}"
}
```
